File: Bronze1-1/06-22-am/unit05/same-digit-substring/samedigit.cpp
/*
 * Same digit substring
 */
#include <bits/stdc++.h>
using namespace std;

int main() {
	string str; cin >> str; str += ' ';
	
	int maxLen = 1;
	int loc = 0;
	for(int j=0; j<str.length();) {
		int k = j+1;
		while(k<str.length() && str[k]==str[j])
			k++;
		//k is the first index so that str[k] != str[j]
		
		if(k-j>maxLen) {
			maxLen = k-j;
			loc = j;
		}
		else if(k-j==maxLen && str[j]>str[loc]) {
			loc = j;
		}
		j = k;
	}
	cout << str.substr(loc, maxLen) << '\n';
}

File: Bronze1-1/06-22-am/unit05/same-digit-substring/samedigit.java
import java.util.*;
public class samedigit {
	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		String str = in.next() + ' ';
		in.close();
		
		int maxLen = 1;
		int loc = 0;
		for(int j=0; j<str.length();) {
			int k = j+1;
			while(k<str.length() && str.charAt(k)==str.charAt(j))
				k++;
			//k is the first index so that str[k] != str[j]
			
			if(k-j>maxLen) {
				maxLen = k-j;
				loc = j;
			}
			else if(k-j==maxLen && str.charAt(j)>str.charAt(loc)) {
				loc = j;
			}
			j = k;
		}
		System.out.println(str.substring(loc, loc+maxLen));
	}
}